blocks/product-gallery-block/features/product-gallery/_children/ProductFocusView/index.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState, useRef } from "react";
+
+import {
+	Button,
+	Icon,
+	Image,
+	Paragraph,
+	Stack,
+	imageANSToImageSrc,
+} from "@wpmedia/arc-themes-components";
+
+const BLOCK_CLASS_NAME = "b-product-gallery";
+
+const ProductFocusView = ({
+	initialItemId,
+	onClose,
+	productImages,
+	resizerAppVersion,
+	resizerURL,
+}) => {
+	const thumbnailBarItemLimit = 3;
+	const getImageIndexById = (id) => productImages.findIndex((image) => image._id === id);
+	const mainImagesRef = useRef([]);
+	const [selectedImageIndex, setSelectedImageIndex] = useState(getImageIndexById(initialItemId));
+	const [thumbnailBarStartIndex, setThumbnailBarStartIndex] = useState(selectedImageIndex);
+
+	const shouldShowDownButton = () =>
+		productImages.length >= thumbnailBarItemLimit &&
+		thumbnailBarStartIndex + thumbnailBarItemLimit < productImages.length;
+
+	const shouldShowUpButton = () => thumbnailBarStartIndex !== 0;
+
+	useEffect(() => {
+		let scrollTimeout;
+		const scrollImageIntoView = () => {
+			const currentImageContainer = mainImagesRef.current?.[selectedImageIndex];
+			if (currentImageContainer?.querySelector("img").complete) {
+				currentImageContainer.scrollIntoView();
+			} else {
+				scrollTimeout = setTimeout(scrollImageIntoView, 10);
+			}
+		};
+		scrollImageIntoView();
+		return () => clearTimeout(scrollTimeout);
+	}, [selectedImageIndex, mainImagesRef]);
+
+	return (
+		<div className={`${BLOCK_CLASS_NAME}__focus-view`}>
+			<Stack direction="horizontal" justifacation="start">
+				<div className={`${BLOCK_CLASS_NAME}__focus-view-thumbnail-bar`}>
+					<Paragraph>
+						{selectedImageIndex + 1} / {productImages.length}
+					</Paragraph>
+					{shouldShowUpButton() ? (
+						<Button
+							accessibilityLabel="Previous Image"
+							onClick={() => setThumbnailBarStartIndex(thumbnailBarStartIndex - 1)}
+						>
+							<Icon name="ChevronUp" />
+						</Button>
+					) : (
+						<div className={`${BLOCK_CLASS_NAME}__focus-view-thumbnail-bar-spacer`} />
```

Comment:
Instead of a spacer, why don't we make this container(`focus-view-thumbnail-bar`) a grid and set the row height?  The "padding" could be row 1, selected text could be row 2, button in row 3, stack in row 4, button in row 5, and the bottom padding is row 6.

```
product-gallery-focus-view-thumbnail-bar: (
    display: grid,
    grid-template-rows: var(--global-spacing-7) auto 1fr auto 1fr var(--global-spacing-7),
    grid-template-columns: var(--global-spacing-6) auto var(--global-spacing-6),
)
```